File: .pub-cache/hosted/pub.dartlang.org/audioplayers-0.20.1/lib/src/logger.dart
import '../audioplayers.dart';
import 'api/log_level.dart';

class Logger {
  static LogLevel _logLevel = LogLevel.ERROR;

  static LogLevel get logLevel => _logLevel;

  static Future<int> changeLogLevel(LogLevel value) {
    _logLevel = value;
    return AudioPlayer.invokeMethod(
      'changeLogLevel',
      <String, dynamic>{'value': value.toString()},
    );
  }

  Logger._() {
    throw UnimplementedError();
  }

  static void log(LogLevel level, String message) {
    if (level.getLevel() <= logLevel.getLevel()) {
      print(message);
    }
  }

  static void info(String message) => log(LogLevel.INFO, message);

  static void error(String message) => log(LogLevel.ERROR, message);
}

File: .pub-cache/hosted/pub.dartlang.org/audioplayers_linux-1.0.1/linux/Logger.cc
#include "Logger.h"

using namespace std;

LogLevel Logger::logLevel = LogLevel::Error;

void Logger::Info(std::string message) {
    Log(LogLevel::Info, message);
}

void Logger::Error(std::string message) {
    Log(LogLevel::Error, message);
}

void Logger::Log(LogLevel level, std::string message) {
    if (level <= logLevel) {
        cout << "AudioPlayers: " << message << endl;
    }
}